CONTRIBUTING.md

```diff
@@ -0,0 +1,97 @@
+
+`main`
+    - Stable branch
+    - Merges come only from `develop` or a hotfix branch (i.e., when something needs to be fixed in "production")
+
+`develop`
+    - Unstable development branch
+    - Last "stable" is tagged: tag before attempting to break something
+
+`feature/<describe-feature>` (including issue id when exists: `feature/62-improve-broken-thing`/`feature/42-add-new-thing`)
+    - Branch from last tag on `develop`
+
+`hotfix/<describe-hotfix>` (including issue id when exists: `hotfix/62-fix-broken-thing-in-release`)
+    - Branch from last tag on `main`
+
+### Pull requests
+
+- As soon as you have made 1 commit in a feature branch, put up a *draft* pull request
+- Keep pull requests small :skull: 
+- :warning: No pre-emptive reviews on PR drafts, **unless** the PR author @-mentions with this specific request
+- When you think you're done, mark PR ready for review, see below (Merge Process)
+
+## Merge Process (into `develop`)
+
+- Create PR from `feature/...` against `develop` (PR author)
+- Describe work in initial comment (PR author)
+    - Reference any related issues (use, e.g., `Fixes #n` or `- Related: #n`)
+    - What does the new code do?
+    - Optional: What should reviewers look at specifically
+    - Information on how to review:
+        - E.g.
+          ```bash
+          pip install ./
+          pytest test/
+          ```
+- Request review (PR author)
+    - Eligible reviewers:
+        - Python code: @led02, @sdruskat, @jkelling
+        - Documentation:
+            - Workflow: @all
+            - Project: @all
+- Review (at least 1 reviewer)
+    - Follow instructions in PR
+    - Review thoroughly beyond instructions
+    - Comments / change suggestions inline in files
+    - Submit review:
+        - **CASE 1:** Non-blocking change requests (typos, documentation wording, etc.) -> Document and Accept
+        - **CASE 2:** Blocking change requests (something doesn't work, bad quality code, docs not understandable):
+            - Ideally, fix things yourself in the branch -> Request Changes (or do them on your own)
+        - **CASE 3:** "Just a comment"s, pointers to potential future changes -> Document and Accept
+        - **CASE 4:** :warning: If you want a second pair of eyes on the PR, use "Comment" to finish review, request another reviewer and @-mention in comment.
+    - Optional: If you find something blocking after initial review, add review with "Request changes" outcome
+- Act on review (PR author)
+    - React to comments
+    - Fix issues
+    - Discuss options
+- Then:
+    - CASE 1: PR author to fix non-blocking issues and merge
+    - CASE 2: PR author to re-request review from original reviewer(s)
+    - CASE 3: PR author to react to comments and merge
+    - CASE 4: (PR author's reaction depends on outcome of second review)
+- Re-review:
+    - See Review above
+- PR author
+    - Close PR, or
+    - Merge PR and delete remote branch
```

Comment:
Should the author merge, or should we not rather require a reviewer to press the button?

Reply:
Also: external contributor cannot merge, so there needs to be a process for this case either way.

Reply:
Either way, I don't care much, but it should be crystal clear who should merge, otherwise we will end up in standoffs with PRs remaining open until someone dares to push the button ;).

Good points though, I'll adapt.